### PR TITLE
getopt added

### PR DESCRIPTION
**Pull request by MohmadAyman**
considering Issue #3355, this is the lib needed for the avr.

### CODE REVIEW

**Comment by Kijewski** [comment]
Please implement the API of `getopt_r()`
```c
int getopt_r(int argc, char *const *argv, const char *optstring, struct opt_r *opt);
```
where `struct opt_r` contains the local variables your algorithm needs and is `{0}` at the first invocation.

**Comment by MohmadAyman** [comment]
Do you mean sth like this ?
```
struct type {
    char *optarg;
    int optind;
    char *scanptr;
} opt_r = { .optarg = NULL, .optind = 0, .scanptr = NULL };
```

**Comment by MohmadAyman** [comment]
ok i've implemented the API, and i'm currently documenting to get over the doxgyen warnings, but just a question, i just don't understand why did you want to this API -where there is a struct containing the local variables- ?.
 i've done it anyway.

File: cpu/atmega_common/getopt.c [hunk 1]
```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (C) 2015 Kaspar Schleiser <Mohmmad.Salah94@eng-st.cu.edu.eg>
```
Comment (cgundogan) [inline cpu/atmega_common/getopt.c]: Use your own name here (:

File: cpu/atmega_common/getopt.c [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+#include <stdio.h>
+#include <string.h>
+
+
+char *optarg;
+int optind = 0;
+char *scanptr = NULL;
+int getopt(int argc, char *argv[], char *optstring)
+{
+char *place;
```
Comment (cgundogan) [inline cpu/atmega_common/getopt.c]: indentation is off here and for the following lines

File: cpu/atmega_common/include/getopt.h [hunk 3]
```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright (C) 2015 Kaspar Schleiser <Mohmmad.Salah94@eng-st.cu.edu.eg>
```
Comment (cgundogan) [inline cpu/atmega_common/include/getopt.h]: adjust the name here, too

File: cpu/atmega_common/include/getopt.h [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     cpu_atmega_common
+ * @{
+ *
+ * @file		header
```
Comment (cgundogan) [inline cpu/atmega_common/include/getopt.h]: indentation

**Comment by OlegHahm** [comment]
Please state a reason for closing.

**Comment by MohmadAyman** [comment]
i was trying a rebase and this happened, since i could solve it, i thought it would be better to close, if there is a way to fix I'll reopen it.

**Comment by OlegHahm** [comment]
Thanks!

**Pull request closed without merging**